src/components/PopupWithForm.js

```diff
@@ -1,10 +1,11 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
+  constructor(popupSelector, handleFormSubmit, submitType) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
+    this._submitType = submitType;
   }
 
   getInputValues() {
```

Comment:
I saw that you complained about `getInputValues` in the 8th sprint.  

Let me explain.  My remark was: Only method `_getInputValues` collects all the data of the inputs and passes it to the call of `submitForm`. So you need to use the collected data rather than the inputs. 

So, it means you need to call `_getInputValues` in line 33 and get the values from the call in `index.js` (rather than make it **public** and call it in index.js) 


```js

      this._handleFormSubmit(this._getInputValues());

```


But it's possible to change `getInputValues` into a **public** method. It's acceptable